### PR TITLE
Create ProtectedRoute and UnauthorizedError components

### LINKED ISSUE #28: FE/IN: Protected Route
\-	Create a superset of Route and add in an authentication check. Please utilise proper types from react-router when creating this component.

Resource: <https://dev.to/mychal/protected-routes-with-react-function-components-dh>

React Router Types: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react-router/index.d.ts

### PR DESCRIPTION
**Pull request by MarianaES**
### What this PR does (required):
- Create the protected route component which verifies if the user is logged in in order to give him/her access to the selected endpoint.
- If the user is not authenticated, an error message will be displayed. 

### Screenshots / Videos (front-end only):
https://user-images.githubusercontent.com/29843870/138613612-b16a9093-13ff-467c-aea4-28cf48a51c60.mov

### Any information needed to test this feature (required):
- Try to access a protected route (i.e. /dashboard, /profile), you'll receive an error.

### CODE REVIEW

File: client/src/components/ProtectedRoute.tsx [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import { ReactElement } from 'react';
+import { Redirect, Route, RouteProps } from 'react-router-dom';
+import { useAuth } from '../context/useAuthContext';
+
+const ProtectedRoute = ({ ...routeProps }: RouteProps): ReactElement => {
+  const { loggedInUser } = useAuth();
+
+  if (!loggedInUser) {
+    return <Redirect to="/unauthorized" />;
```
Comment (sundayezeilo) [inline client/src/components/ProtectedRoute.tsx:9]: :+1:

**Review by sundayezeilo — COMMENTED** [review]
This one looks good :+1:

File: client/src/App.tsx [hunk 2]
```diff
@@ -20,31 +22,32 @@ import './App.css';
...
               <NavBar />
               <Switch>
                 <Route exact path="/login" component={Login} />
                 <Route exact path="/signup" component={Signup} />
-                <Route path="/profile" component={Profile} />
-                <Route exact path="/my-jobs" component={MyJobs} />
-                <Route exact path="/my-sitters" component={MySitters} />
-                <Route exact path="/messages" component={Messages} />
-                <Route exact path="/dashboard">
+                <ProtectedRoute path="/profile" component={Profile} />
```
Comment (rajivtitus) [inline client/src/App.tsx:36]: 👍🏽

File: client/src/components/UnauthorizedError/UnauthorizedError.tsx [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import Alert from '@material-ui/lab/Alert';
+import AlertTitle from '@material-ui/lab/AlertTitle';
+
+function UnauthorizedError(): JSX.Element {
```
Comment (rajivtitus) [inline client/src/components/UnauthorizedError/UnauthorizedError.tsx:4]: 💯

**Review by rajivtitus — APPROVED** [review]
Great work here! Nice touch adding the unauthorized error alert 🚀

**Pull request merged**